src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/CompileItemHandler.cs

```diff
@@ -31,6 +30,7 @@ public CompileItemHandler(UnconfiguredProject project)
         }
 
         [ImportMany]
+        [NotNull, AllowNull]
         private readonly IEnumerable<IFileRenameHandler> _fileRenameHandlers = null;
```

Comment:
This one was interesting. Here `AllowNull` says the field can be set to null (as done in the initializer), yet `NotNull` says that consumers of the field will not receive a null. In this case that's true enough as the property's set by MEF after the ctor completes.

Reply:
Should this set of attributes be added for all MEF import collections? Is this the general pattern we want to use?

Reply:
I'm on the fence here. It works, and is a 'correct' usage of the attributes, but it looks pretty weird.

An alternative is just to assign an empty array here. It involves more work/allocation than just assigning null (which is a no-op) but the overhead is trivial and it's easier to see that it's correct.

The way it is now, it's valid to set it to null elsewhere in the class, which is obviously not what we want. I think I'll assign `Array.Empty<T>()` here, and would vote for doing that elsewhere too.